packages/ott-vis-panel/src/components/TreeDisplay.tsx

```diff
@@ -0,0 +1,317 @@
+			const balancerCircles = balancerGroup.selectAll(".balancer").data(balancerNodes);
+			balancerCircles
+				.enter()
+				.append("circle")
+				.attr("class", "balancer")
+				.attr("r", NODE_RADIUS)
+				.attr("fill", d => color(d.group))
+				.attr("stroke", "white")
+				.attr("stroke-width", 2)
+				.attr("cx", d => d.x)
+				.attr("cy", d => d.y)
+				.attr("data-nodeid", d => d.id);
+			balancerCircles.exit().remove();
+
+			// create groups for all the monoliths
+			const monolithGroup = wholeGraph.select("g.monoliths");
+			const monolithGroups = monolithGroup.selectAll(".monolith").data(monolithNodes);
+			monolithGroups
+				.enter()
+				.append("g")
+				.attr("class", "monolith")
+				.attr("transform", (d, i) => `translate(${d.x}, ${d.y})`)
+				.each(function (d) {
+					const diagonal = d3
+						.linkHorizontal<any, TreeNode>()
+						.x((d: any) => d.y)
+						.y((d: any) => d.x);
+
+					const monolith = d3.select(this);
+					const monolithLinks = monolith.selectAll(".treelink").data(d.tree.links());
+					monolithLinks
+						.enter()
+						.append("path")
+						.attr("class", "treelink")
+						.attr("d", diagonal)
+						.attr("fill", "none")
+						.attr("stroke", "white")
+						.attr("stroke-width", 1.5)
+						.attr("data-nodeid-source", d => d.source.data.id)
+						.attr("data-nodeid-target", d => d.target.data.id);
+					monolithLinks.exit().remove();
+
+					const monolithCircles = monolith
+						.selectAll(".monolith")
+						.data(d.tree.descendants());
+					monolithCircles
+						.enter()
+						.append("circle")
+						.attr("class", "monolith")
+						.attr("r", d => radius(d.data))
+						.attr("fill", d => color(d.data.group))
+						.attr("stroke", "white")
+						.attr("stroke-width", 2)
+						.attr("cx", (d: any) => d.y)
+						.attr("cy", (d: any) => d.x)
+						.attr("data-nodeid", d => d.data.id);
+					monolithCircles.exit().remove();
+				});
+
+			// create the links between balancers and monoliths
+			interface B2MLink {
+				source: BalancerNode;
+				target: MonolithNode;
+			}
+			const diagonal = d3
+				.linkHorizontal<B2MLink, BalancerNode | MonolithNode>()
+				.x(d => d.x)
+				.y(d => d.y);
+
+			const b2mLinkData = balancerNodes.flatMap(balancer => {
+				return monolithNodes.map(monolith => {
+					return {
+						source: balancer,
+						target: monolith,
+					};
+				});
+			});
+			const balancerMonolithLinks = gb2mLinks.selectAll(".b2m-link").data(b2mLinkData);
+			balancerMonolithLinks
+				.enter()
+				.append("path")
+				.attr("class", "b2m-link")
+				.attr("d", diagonal)
+				.attr("fill", "none")
+				.attr("stroke", "white")
+				.attr("stroke-width", 1.5)
+				.attr("data-nodeid-source", d => d.source.id)
+				.attr("data-nodeid-target", d => d.target.id);
+			balancerMonolithLinks.exit().remove();
+
+			let zoom = d3.zoom<SVGSVGElement, TreeNode>().on("zoom", handleZoom);
+			function handleZoom(e: any) {
+				d3.select("g.chart").attr("transform", e.transform);
```

Comment:
to avoid conflicting with other panels

```suggestion
				svg.select("g.chart").attr("transform", e.transform);
```